### PR TITLE
chore(analytics): error stack trace metadata error handling and wrapped errors in analytics

### PR DESCRIPTION
**Pull request by mkhq**
**What this PR does / why we need it**:

Updates error handling in getStackTraceMetadata and introduces support for wrappedErrors.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: core/src/exceptions.ts [hunk 1]
```diff
@@ -183,29 +198,35 @@ export function formatGardenErrorWithDetail(error: GardenError) {
...
+    const filePath = atLine[2]
+    let lastSrcFilePos = -1
+    let tmpPos = -1
+
+    if ((tmpPos = filePath.lastIndexOf("src")) > -1) {
+      lastSrcFilePos = tmpPos + 4
+    } else if ((tmpPos = filePath.lastIndexOf("node_modules")) > -1) {
+      lastSrcFilePos = tmpPos + 13
+    } else if ((tmpPos = filePath.lastIndexOf("node:internal")) > -1) {
+      lastSrcFilePos = tmpPos + 14
```
Comment (shumailxyz) [inline core/src/exceptions.ts:222]: Some comments using constant values e.g. 4,13,14 here would be nice.

would be handy to read in future :)

File: core/src/exceptions.ts [hunk 2]
```diff
@@ -214,8 +235,25 @@ export function getStackTraceMetadata(error: GardenError): GardenErrorStackTrace
       },
     ]
   })
+}
+
+export function getStackTraceMetadata(error: GardenError): GardenErrorStackTrace {
+  if (!error.stack && !error.wrappedErrors) {
+    return { metadata: [], wrappedMetadata: undefined }
```
Comment (shumailxyz) [inline core/src/exceptions.ts:242]: I feel like we should wrap the following block in try/catch as well and just return like this in catch
```
return { metadata: [], wrappedMetadata: undefined }
```

**Review by shumailxyz — APPROVED** [review]
some non blocking nits that would be nice to address.

File: core/src/exceptions.ts (hunk 2 above)
Comment (vvagaytsev) [reply core/src/exceptions.ts:242]: Agree, also we can log something if an error occurs in the `try`-block.

Comment (mkhq) [reply core/src/exceptions.ts:248]: Wrapped the call points in analytics.ts with try catch since we can log the errors there.

**Review by shumailxyz — APPROVED** [review]
Looks good to me ✅ Thank you.

**Comment by shumailxyz** [comment]
Also, lets squash while merging.

**Pull request merged**